### PR TITLE
POC version for scalecube-js

### PR DESCRIPTION
**Pull request by idanilt**
POC version for scalecube js
Implemented only the Microservices basic pattern only local services without remote services (no gossip or SWIM)
First implementation, we be restructured to something normal

### CODE REVIEW

File: src/scalecube-services/api/Microservices.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+}
+export interface ServiceCall {
+  invoke(Message: message): Promise<Message>;
+  // listen(Message: message): Observable<Message>; TBD might use Iterators instead
+}
+export interface DispatcherContext {
+  create(): ServiceCall;
+}
+export interface ProxyContext {
```
Comment (ronenhamias) [inline src/scalecube-services/api/Microservices.js]: i dont think we need proxy context as interface

File: src/scalecube-services/api/Microservices.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  serviceLookup(serviceName: string): List<ServiceInstance> ;
+  getLocalInstance(serviceName: string, method:string): Optional<ServiceInstance>;
+  services(): Collection<ServiceInstance>;
+  getServiceDefinition(serviceName: string): Optional<ServiceDefinition>;
+  registerInterface<T>(serviceInterface: T): ServiceDefinition;
+}
+export interface Router {
+  route(message:Message): ServiceInstance;
+}
+export interface Builder{
```
Comment (ronenhamias) [inline src/scalecube-services/api/Microservices.js]: i dont think we need builder as interface

File: src/scalecube-services/Router.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// @flow
+export interface Router {
```
Comment (ronenhamias) [inline src/scalecube-services/Router.js]: i think router should be under 'services/api/Router'

File: src/scalecube-services/services/ServiceCall.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+export class ServiceCall{
+  router: Router;
+  constructor(router: Router, timeout:number = 5000){
+    this.router = router;
+  }
+  invoke(message:any) {
+    const inst = this.router.route(message);
+    if( inst ) {
+      return inst.service[message.method](...message.data);
+    }
```
Comment (ronenhamias) [inline src/scalecube-services/services/ServiceCall.js]: if service is not found on else statement need to throw error
service not found error.

Comment (idanilt) [reply src/scalecube-services/services/ServiceCall.js]: It gives a warnning
I can change it to error
```javascript
invoke(message:any) {
    const inst = this.router.route(message);
    if( inst ) {
      return inst.service[message.method](...message.data);
    }
    console.warn(`Service not found for message ${message}`);
  }
```

**Comment by ronenhamias** [comment]
```javascript
 it('Dispatcher should fail with service not found error', ()=>{

        const microservices = Microservices.builder().build();
        const dispatcher = microservices.dispatcher().create();

        const message = {
            serviceName: 'GreetingService',
            method: 'hello',
            data: 'Idan'
        };

        try {
            dispatcher.invoke(message);
        } catch(ex) {
            console.log(ex);
            expect(ex.message).toBe("Service not found error: GreetingService.hello");
        }

    });

    it('Dispatcher should fail with service not found error', ()=>{

        const microservices = Microservices.builder().build();
        const dispatcher = microservices.dispatcher().create();

        const message = {
            serviceName: 'GreetingService',
            method: 'hello',
            data: 'Idan'
        };

        try {
            dispatcher.invoke(message);
        } catch(ex) {
            expect(ex.message).toBe("Service not found error: GreetingService.hello");
        }
    });
```

**Pull request merged**